internal/cmd/skupper/link/link.go

```diff
@@ -15,15 +19,108 @@ func NewCmdLink() *cobra.Command {
 skupper link status`,
 	}
 
-	linkExportCommand := kube.NewCmdLinkGenerate()
-	linkUpdateCommand := kube.NewCmdLinkUpdate()
-	linkStatusCommand := kube.NewCmdLinkStatus()
-	linkDeleteCommand := kube.NewCmdLinkDelete()
+	cmd.AddCommand(CmdLinkGenerateFactory(config.GetPlatform()))
+	cmd.AddCommand(CmdLinkUpdateFactory(config.GetPlatform()))
+	cmd.AddCommand(CmdLinkStatusFactory(config.GetPlatform()))
+	cmd.AddCommand(CmdLinkDeleteFactory(config.GetPlatform()))
 
-	cmd.AddCommand(&linkExportCommand.CobraCmd)
-	cmd.AddCommand(&linkUpdateCommand.CobraCmd)
-	cmd.AddCommand(&linkDeleteCommand.CobraCmd)
-	cmd.AddCommand(&linkStatusCommand.CobraCmd)
+	return cmd
+}
+
+func CmdLinkGenerateFactory(configuredPlatform types.Platform) *cobra.Command {
+	kubeCommand := kube.NewCmdLinkGenerate()
+	nonKubeCommand := non_kube.NewCmdLinkGenerate()
+
+	cmdLinkGenerateDesc := common.SkupperCmdDescription{
+		Use:   "generate",
+		Short: "Generate a new link resource in a yaml file",
+		Long: `Generate a new link resource with the data needed from the target site. The resultant
+output needs to be applied in the site in which we want to create the link.`,
+	}
+
+	cmd := common.ConfigureCobraCommand(configuredPlatform, cmdLinkGenerateDesc, kubeCommand, nonKubeCommand)
+	cmdFlags := common.CommandLinkGenerateFlags{}
+	cmd.Flags().StringVar(&cmdFlags.TlsSecret, common.FlagNameTlsSecret, "", common.FlagDescTlsSecret)
+	cmd.Flags().StringVar(&cmdFlags.Cost, common.FlagNameCost, "1", common.FlagDescCost)
+	cmd.Flags().StringVarP(&cmdFlags.Output, common.FlagNameOutput, "o", "yaml", common.FlagDescOutput)
+	cmd.Flags().BoolVar(&cmdFlags.GenerateCredential, common.FlagNameGenerateCredential, true, common.FlagDescGenerateCredential)
+	cmd.Flags().StringVar(&cmdFlags.Timeout, common.FlagNameTimeout, "60", common.FlagDescCost)
```

Comment:
wrong description for timeout command

Reply:
Thanks!! fixed: 8e73d493cba3186ef9e046b6a03bf3f266d63869